models/models.js

```diff
@@ -54,3 +54,21 @@ exports.selectCommentsByArticleId = (id) => {
         });
 };
 
+exports.insertComment = (newComment) => {
+    const { author, body, article_id} = newComment
+    return db.query(`INSERT INTO comments
+                     (author, body, article_id)
+                     VALUES
+                     ($1, $2, $3) RETURNING *`,
+                     [author, body, article_id])
+    .then(({rows}) => {
+        return rows[0]
+    })
+};
+
+exports.checkAuthorExists = (username) => {
+    return db.query(`SELECT * FROM users WHERE username = $1`, [username])
+        .then(({ rows }) => {
+                return rows.length > 0
+        });
+    }
```

Comment:
You shouldn't need to manually check that the user exists. You should receive an SQL error if you try to insert a comment with a user which doesn't exist. 

Please handle it in your SQL error handler if possible.